include/hipSYCL/sycl/libkernel/accessor.hpp

```diff
@@ -537,15 +537,17 @@ class accessor : public detail::accessor_base<std::remove_const_t<dataT>> {
   }
 
   template<int D = dimensions,
-            std::enable_if_t<(D > 0), bool> = true>
+            access::mode M = accessmode,
+            std::enable_if_t<(D > 0) && (M != access::mode::atomic), bool> = true>
   HIPSYCL_UNIVERSAL_TARGET
   reference operator[](id<dimensions> index) const
   {
     return (this->_ptr.get())[detail::linear_id<dimensions>::get(index, _buffer_range)];
   }
 
   template<int D = dimensions,
-            std::enable_if_t<(D == 1), bool> = true>
+            access::mode M = accessmode,
+            std::enable_if_t<(D == 1) && (M != access::mode::atomic), bool> = true>
```

Comment:
Wouldn't local accessor be affected as well?

Reply:
only if they had atomic support in the first case (which they [deprecatedly should have as of SYCL 2020](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_accessor_specialization_with_targetlocal))